cmd/clusterctl/client/cluster/workload_cluster.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	kc "sigs.k8s.io/cluster-api/util/kubeconfig"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WorkloadCluster has methods for fetching kubeconfig of workload cluster from management cluster.
+type WorkloadCluster interface {
+	//Get workload cluster kubeconfig
+	GetKubeconfig(workloadClusterName string, namespace string) (string, error)
+}
+
+// workloadCluster implements WorkloadCluster.
+type workloadCluster struct {
+	proxy Proxy
+}
+
+func (p *workloadCluster) GetKubeconfig(workloadClusterName string, namespace string) (string, error) {
+	cs, err := p.proxy.NewClient()
+	if err != nil {
+		return "", err
+	}
+
+	obj := client.ObjectKey{
+		Namespace: namespace,
+		Name:      workloadClusterName,
+	}
+	dataBytes, err := kc.FromSecret(ctx, cs, obj)
+
+	data := string(dataBytes)
+	return data, err
```

Comment:
Could we wrap the error we return here with the namespace?

If I create a cluster `wff-test-3` in namespace `foobar`, when I run `clusterctl get kubeconfig wff-test-3` I get the following error `Error: secrets "wff-test-3-kubeconfig" not found`. 

I suggest returning an error that may convey to the user some more information. 
```
Error: "wff-test-3-kubeconfig` not found in namespace "default"
```
Also I don't think we need to show that it is a secret object we are looking for but that is just my opinion.

Reply:
Agreed. Have updated accordingly.